platforms/android/tangram/src/main/cpp/platform_android.cpp

```diff
@@ -350,6 +373,30 @@ void setCurrentThreadPriority(int priority) {
     setpriority(PRIO_PROCESS, tid, priority);
 }
 
+void sceneUpdateCallback(jobject updateStatusCallbackRef, const std::vector<Tangram::SceneUpdateStatus>& sceneUpdateStatus) {
+
+    if (!updateStatusCallbackRef) {
+        return;
+    }
+
+    JniThreadBinding jniEnv(jvm);
+
+    jobject arrayList = jniEnv->NewObject(arrayListClass, arrayListInitMID);
+
+    for (auto updateStatus : sceneUpdateStatus) {
+        jstring jUpdateStatusPath = jniEnv->NewStringUTF(updateStatus.update.path.c_str());
+        jstring jUpdateStatusValue = jniEnv->NewStringUTF(updateStatus.update.value.c_str());
+        jint jError = (jint)updateStatus.error;
+        jobject jUpdateStatus = jniEnv->NewObject(sceneUpdateStatusClass, sceneUpdateStatusInitMID,
+                                                  jUpdateStatusPath, jUpdateStatusValue, jError);
+
+        jniEnv->CallBooleanMethod(arrayList, arrayListAddMID, jUpdateStatus);
+    }
+
+    jniEnv->CallVoidMethod(updateStatusCallbackRef, onSceneUpdateErrorMID, arrayList);
```

Comment:
You could have created a method on which takes in 2 strings and a int and created the array list on java itself... but maybe what you currently have has less handshakes with jni interface.. Just a thought.